### PR TITLE
Override combobox accessability

### PR DESCRIPTION
**Pull request by mXaln**
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/537)

### CODE REVIEW

File: jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                oldValue = this.value
+                it.consume()
+            }
+            KeyCode.ESCAPE -> {
+                this.value = oldValue
+                it.consume()
+            }
+        }
+    }
+
```
Comment (AnonymousWalker) [inline jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt:93]: should we call consume() just once after capture/filtering the event? Maybe at line 93 and 113

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/537)* status: 0 of 12 files reviewed, 1 unresolved discussion (waiting on @AnonymousWalker)

___
*[jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt, line 93 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/537#-MzQX0_orZVcLaKgR_iZ-r1-93:-MzV9kl2BGN13q9nDOSW:b3kk4js) ([raw file](https://github.com/bible-translation-tools/orature/blob/53fffe3977f7f232866819ce22dfae210ea59f49/jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt#L93)):*
<details><summary><i>Previously, AnonymousWalker (Tony) wrote…</i></summary><blockquote>

should we call consume() just once after capture/filtering the event? Maybe at line 93 and 113
</blockquote></details>

Sorry, the line numbers that you specified are different in my editor, for some reason. If you mean to consume all events, then we don't want this. We want to consume only specific key events and let others pass through.

**Review by AnonymousWalker — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/537)* status: 0 of 12 files reviewed, 2 unresolved discussions (waiting on @AnonymousWalker and @mXaln)

___
*[jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt, line 73 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/537#-MzV9_fMD_heSOmkF7wG:-MzV9_fMD_heSOmkF7wH:b8fyqvb) ([raw file](https://github.com/bible-translation-tools/orature/blob/53fffe3977f7f232866819ce22dfae210ea59f49/jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt#L73)):*
> ```Kotlin
>  * @param action Action to invoke when value has changed
>  */
> fun <T> ComboBox<T>.overrideEventsAndRun(action: (T) -> Unit = {}) {
> ```

looks like this method capture the keyboard events, could you name it something more specific? ` overrideEvents ` sounds like overriding ALL event this node could potentially emit

**Comment by AnonymousWalker** [comment]
*[jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt, line 93 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/537#-MzQX0_orZVcLaKgR_iZ-r1-93:-MzVBgalEdP3fThaeul1:b-kett6w) ([raw file](https://github.com/bible-translation-tools/orature/blob/53fffe3977f7f232866819ce22dfae210ea59f49/jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt#L93)):*
<details><summary><i>Previously, mXaln (Maxim) wrote…</i></summary><blockquote>

Sorry, the line numbers that you specified are different in my editor, for some reason. If you mean to consume all events, then we don't want this. We want to consume only specific key events and let others pass through.
</blockquote></details>

thanks for clarifying!

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/537)* status: 0 of 12 files reviewed, 1 unresolved discussion (waiting on @AnonymousWalker)

___
*[jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt, line 73 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/537#-MzV9_fMD_heSOmkF7wG:-MzVCvQS9H7zLMZHwBRT:b-fv3ngl) ([raw file](https://github.com/bible-translation-tools/orature/blob/53fffe3977f7f232866819ce22dfae210ea59f49/jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt#L73)):*
<details><summary><i>Previously, AnonymousWalker (Tony) wrote…</i></summary><blockquote>

looks like this method capture the keyboard events, could you name it something more specific? ` overrideEvents ` sounds like overriding ALL event this node could potentially emit
</blockquote></details>

yeah, I was thinking of different names. And came up with this one. If you have a better idea, would be good.

**Comment by AnonymousWalker** [comment]
*[jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt, line 73 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/537#-MzV9_fMD_heSOmkF7wG:-MzVLTPa7J7ze9PHOeEl:b-3a6se0) ([raw file](https://github.com/bible-translation-tools/orature/blob/53fffe3977f7f232866819ce22dfae210ea59f49/jvm/utils/src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/NodeUtils.kt#L73)):*
<details><summary><i>Previously, mXaln (Maxim) wrote…</i></summary><blockquote>

yeah, I was thinking of different names. And came up with this one. If you have a better idea, would be good.
</blockquote></details>

how about ` overrideDefaultKeyEvent() ` ?

**Review by AnonymousWalker — APPROVED** [review]

**Review by jsarabia — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/537#-:-MzaMB2p7yQ-7auEw8De:bnfp4nl)* status: 0 of 12 files reviewed, 1 unresolved discussion (waiting on @AnonymousWalker and @mXaln)

**Comment by OpThomasPrime1** [comment]
Verified by QA (Thomas Oliver) Windows, Linux, Apple

**Pull request merged**